### ISSUE TITLE
Dgraph accepts tilde character ("~") in the predicate name for mutations

### ISSUE THREAD

**Issue by danielmai** [dgraph-io/dgraph#2568]
## Bug report

**Desktop:**
 - Dgraph Version: v1.0.8

**Describe the bug**

Running a mutation with the tilde character (`~`) in the predicate name creates the predicate with that name.

Dgraph should reject the mutation since "~" is used for reverse edges.

**To Reproduce**

Run 1 Zero, 1 Alpha.

Run this mutation:

```
{
  set {
    _:alice <~isSomething> _:bob .
  }
}
```

See the predicate name `~isSomething` in the schema.

```
curl -s localhost:8080/query -d 'schema {}' | jq '.data.schema[2]'
```

```
{
  "predicate": "~isSomething",
  "type": "uid"
}
```

**Expected behavior**

The mutation is rejected.

**Comment by danielmai** [comment]
Export data looks like the following. Export was done with `curl localhost:8080/admin/export`.

RDF:
```
$ cat dgraph-1-2018-09-06-15-29.rdf
<_:uid2711> <~isSomething> _:uid2712 .
```

Schema:
```
$ cat dgraph-1-2018-09-06-15-29.schema
_share_hash_:string @index(exact) .
~isSomething:uid .
```

**Comment by danielmai** [comment]
Fixed in 98d9a3641bf79cadaae48890af8bdf10d8ba6298.

Trying to run the same mutation in the latest Dgraph now gives an error:

```
{
  "errors": [
    {
      "code": "ErrorInvalidRequest",
      "message": "Key error: predicate \"~isSomething\" has invalid tilde: subject:\"_:alice\" predicate:\"~isSomething\" object_id:\"_:bob\" "
    }
  ],
  "data": null
}
```
